### ISSUE TITLE
vscode module is deprecated

### ISSUE THREAD

**Issue by shadycuz** [nvuillam/vscode-groovy-lint#120]
@nvuillam Hey I was trying to get a development environment setup so I could contribute. 

I couldn't see which version of `npm` to use but I saw in the actions we are using version 12 of `node`. So I installed that and ended up with:

```sh
[~/repos/dsty/vscode-groovy-lint]
shadycuz-> node --version
v12.22.7
[~/repos/dsty/vscode-groovy-lint]
shadycuz-> npm --version
6.14.15
```

I wasn't sure where to start so I tried the steps from the contributing guide to start debugging but that failed:

![image](https://user-images.githubusercontent.com/17090999/137145010-5153e2cb-5d37-4f75-aa56-6e7559dc60d0.png)

![image](https://user-images.githubusercontent.com/17090999/137144830-fe3b5d41-1fd7-43c5-adcd-f6039b42a752.png)

I'm guessing because I didn't run `npm install` 😊 .

So then I tried installing:

```sh
[~/repos/dsty/vscode-groovy-lint]
shadycuz-> npm install

> vscode-groovy-lint@1.4.0 postinstall /home/shadycuz/repos/dsty/vscode-groovy-lint
> cd client && npm install --unsafe-perm && cd ../server && npm install --unsafe-perm && cd ..


> vscode-groovy-lint-client@0.0.1 postinstall /home/shadycuz/repos/dsty/vscode-groovy-lint/client
> vscode-install

Detected VS Code engine version: ^1.42.0
Error installing vscode.d.ts: Error: Request returned status code: 404
Details: <!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /api/releases/stable</pre>
</body>
</html>

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vscode-groovy-lint-client@0.0.1 postinstall: `vscode-install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the vscode-groovy-lint-client@0.0.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/shadycuz/.npm/_logs/2021-10-13T14_05_26_870Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vscode-groovy-lint@1.4.0 postinstall: `cd client && npm install --unsafe-perm && cd ../server && npm install --unsafe-perm && cd ..`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the vscode-groovy-lint@1.4.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/shadycuz/.npm/_logs/2021-10-13T14_05_26_920Z-debug.log
```

When trying to fix this I found this [post](https://github.com/microsoft/vscode/issues/119822#issuecomment-806489621). Which says `vscode` module has been deprecated with 2 options for fixing. Update to at least `vscode@1.1.37` or migrate using this [guide](https://code.visualstudio.com/api/working-with-extensions/testing-extension#migrating-from-vscode). 

So I tried to upgrade `vscode@1.1.37` which actually worked, but I was still unable to run the debugging because when the process host opened I got this.

You can either provide me the `node` and `npm` version you use or we can stick with version `1.1.37` of `vscode` or I can try to follow the steps in the upgrade guide.

**Comment by nvuillam** [comment]
@shadycuz sorry for the delay

did you try to use [VsCode Insiders](https://code.visualstudio.com/insiders/) ?

**Comment by shadycuz** [comment]
I could try it but I'm not sure how that would help 🤔

**Comment by nvuillam** [comment]
I forgot to tell that in CONTRIBUTING.md, that is corrected now :)

**Comment by shadycuz** [comment]
@nvuillam Maybe we are having misunderstanding? I noticed that when you updated the README, the tests [failed](https://github.com/nvuillam/vscode-groovy-lint/runs/3920161175?check_suite_focus=true). I'm pretty sure that:
```
> vscode-groovy-lint-client@0.0.1 postinstall /home/shadycuz/repos/dsty/vscode-groovy-lint/client
> vscode-install

Detected VS Code engine version: ^1.42.0
Error installing vscode.d.ts: Error: Request returned status code: 404
```
is not going to go away if I change to `vs-insiders`. 

In my branch from PR #121, the `Groovy Lint debug` option works fine now that I have updated the version constraint on `vscode` package. 

This is using normal IDE and not `insiders`.

![image](https://user-images.githubusercontent.com/17090999/137854597-5a2494b5-6972-41d8-8a6b-f1176870215f.png)
